jenkins/bootstrap.py

```diff
@@ -119,9 +120,20 @@ def checkout(repo, branch, pull):
     call([git, 'init', repo])
     os.chdir(repo)
     # TODO(fejta): cache git calls
-    call([
-        git, 'fetch', '--tags', repository(repo), ref,
-    ])
+    retries = 3
+    while retries > 0:
+        retries -= 1
+        try:
+            call([
+                git, 'fetch', '--tags', repository(repo), ref,
+            ])
```

Comment:
You need to break if this is successful, either here or in a try-except-ese block.